[2024-12-16T19:32:07.802+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sick_leave_etl_dag.extract_data manual__2024-12-16T19:27:04.574862+00:00 [queued]>
[2024-12-16T19:32:07.808+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sick_leave_etl_dag.extract_data manual__2024-12-16T19:27:04.574862+00:00 [queued]>
[2024-12-16T19:32:07.809+0000] {taskinstance.py:1359} INFO - Starting attempt 2 of 2
[2024-12-16T19:32:07.820+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): extract_data> on 2024-12-16 19:27:04.574862+00:00
[2024-12-16T19:32:07.829+0000] {standard_task_runner.py:57} INFO - Started process 5841 to run task
[2024-12-16T19:32:07.833+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'sick_leave_etl_dag', 'extract_data', 'manual__2024-12-16T19:27:04.574862+00:00', '--job-id', '67', '--raw', '--subdir', 'DAGS_FOLDER/sick_leave_dag.py', '--cfg-path', '/tmp/tmpegfalm40']
[2024-12-16T19:32:07.835+0000] {standard_task_runner.py:85} INFO - Job 67: Subtask extract_data
[2024-12-16T19:32:07.850+0000] {logging_mixin.py:151} WARNING - /home/***/.local/lib/python3.8/site-packages/***/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2024-12-16T19:32:07.881+0000] {task_command.py:415} INFO - Running <TaskInstance: sick_leave_etl_dag.extract_data manual__2024-12-16T19:27:04.574862+00:00 [running]> on host cfbf61ae1f16
[2024-12-16T19:32:07.955+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='sick_leave_etl_dag' AIRFLOW_CTX_TASK_ID='extract_data' AIRFLOW_CTX_EXECUTION_DATE='2024-12-16T19:27:04.574862+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-16T19:27:04.574862+00:00'
[2024-12-16T19:32:08.055+0000] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/sick_leave_dag.py", line 34, in extract_data_from_excel
    df.to_csv('/opt/airflow/dags/data/extracted_data.csv', index=False)  # Сохраняем данные для передачи между тасками
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/core/generic.py", line 3772, in to_csv
    return DataFrameRenderer(formatter).to_csv(
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/formats/format.py", line 1186, in to_csv
    csv_formatter.save()
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/formats/csvs.py", line 240, in save
    with get_handle(
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/common.py", line 737, in get_handle
    check_parent_directory(str(handle))
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/common.py", line 600, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: '/opt/***/dags/data'
[2024-12-16T19:32:08.068+0000] {taskinstance.py:1398} INFO - Marking task as FAILED. dag_id=sick_leave_etl_dag, task_id=extract_data, execution_date=20241216T192704, start_date=20241216T193207, end_date=20241216T193208
[2024-12-16T19:32:08.076+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 67 for task extract_data (Cannot save file into a non-existent directory: '/opt/***/dags/data'; 5841)
[2024-12-16T19:32:08.085+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-12-16T19:32:08.105+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
